nextcord/ext/commands/bot.py

```diff
@@ -901,6 +903,155 @@ def reload_extension(self, name: str, *, package: Optional[str] = None) -> None:
+            raise ValueError("Cannot provide both package and packages.")
+
+        if packages:
+            if len(packages) != len(names):
+                raise ValueError("The length of packages must match the length of extensions.")
+            packages_itr = iter(packages)
+
+        if extras:
+            if len(extras) != len(names):
+                raise ValueError("The length of extra parameters must match the length of extensions.")
+            extras_itr = iter(extras)
+
+        loaded_extensions: List[str] = []
+
+        for extension in names:
+            if packages:
+                # this shouldn't give any errors since we already checked the length
+                cur_package = next(packages_itr)
+
+            if extras:
+                # this shouldn't give any errors since we already checked the length
+                cur_extra = next(extras_itr)
+
+            try:
+                if package:
+                    self.load_extension(extension, package=package, extras=cur_extra)
+                elif packages:
+                    self.load_extension(extension, package=cur_package, extras=cur_extra)
+            except Exception:
+                raise
+            else:
+                loaded_extensions.append(extension)
+
+        return loaded_extensions
+
+    def find_load_extensions(self, folder_name: str, filter: str = "*.py") -> List[str]:
+        """Loads all extensions found in a folder.
+
+        Once an extension found in a folder has been loaded and did not throw
+        any exceptions, it will be added to a list of extension names that
+        will be returned.
+
+        Parameters
+        ----------
+        folder_name: :class:`str`
+            The name (or path) of the folder to look through.
+        filter: :class:`str`
+            The filter to use when looking for extensions.
+
+        Returns
+        --------
+        A :type:`List[str]` that contains the names of all of the extensions
+        that loaded successfully.
+
+        Raises
+        --------
+        ExtensionNotFound
+            An extension could not be imported.
+        ExtensionAlreadyLoaded
+            An extension is already loaded.
+        NoEntryPointError
+            An extension does not have a setup function.
+        ExtensionFailed
+            An extension or its setup function had an execution error.
+        """
+        path = Path(folder_name)
+        _raw_files: List[Path] = list(path.rglob(filter))
+
+        packages: List[str] = [str(p.parent).replace(os.sep, ".") for p in _raw_files]
+        modules: List[str] = [str(p.stem) for p in _raw_files]
+        extensions: List[str] = self.load_extensions(modules, packages=packages)
+
+        return extensions
+
+    def find_load_module_extensions(self, folder_name: str, suffix: str = ".py") -> List[str]:
+        """Similar to :meth:`find_load_extensions` except it filters out any files that aren't module init files.
```

Comment:
Whats the use case for this method?

Reply:
So if you have a certain cog setup where each cog has their own folder and they have multiple Python files inside. I added it because Denver suggested it.

Reply:
Sure, but your current implement doesn't do what you just described. All it does is recursively attempt to load all ``__init__.py`` files. And it doesn't appear to be properly recursive either when tested locally. Using the method simply yields the first found ``__init__.py`` file when viewing the returned data.